src/components/Banner/Banner.tsx

```diff
@@ -47,135 +37,127 @@ export interface Props {
   onDismiss?(): void;
 }
 
-export type CombinedProps = Props & WithContextTypes<WithinContentContextType>;
-
-export class Banner extends React.PureComponent<CombinedProps, never> {
-  render() {
-    const {
-      icon,
-      action,
-      secondaryAction,
-      title,
-      children,
-      status,
-      onDismiss,
-      context: {withinContentContainer},
-    } = this.props;
-
-    let color: IconProps['color'];
-    let defaultIcon: IconProps['source'];
-    let ariaRoleType = 'status';
-
-    switch (status) {
-      case 'success':
-        color = 'greenDark';
-        defaultIcon = CircleTickMajorTwotone;
-        break;
-      case 'info':
-        color = 'tealDark';
-        defaultIcon = CircleInformationMajorTwotone;
-        break;
-      case 'warning':
-        color = 'yellowDark';
-        defaultIcon = CircleAlertMajorTwotone;
-        ariaRoleType = 'alert';
-        break;
-      case 'critical':
-        color = 'redDark';
-        defaultIcon = CircleDisabledMajorTwotone;
-        ariaRoleType = 'alert';
-        break;
-      default:
-        color = 'inkLighter';
-        defaultIcon = FlagMajorTwotone;
-    }
-    const className = classNames(
-      styles.Banner,
-      status && styles[variationName('status', status)],
-      onDismiss && styles.hasDismiss,
-      withinContentContainer
-        ? styles.withinContentContainer
-        : styles.withinPage,
-    );
+export default function Banner({
+  icon,
+  action,
+  secondaryAction,
+  title,
+  children,
+  status,
+  onDismiss,
+}: Props) {
+  const {withinContentContainer} = React.useContext(WithinContentContext);
```

Comment:
A lot of red for not many changes. It's mostly class -> function with the removal of HoC and this addition

Reply:
Diff settings -> "hide whitespace changes" shall be our new best friend for a lot of these convert to function versions

Reply:
Definitely 🙏 , luckily we don't have many components to transform 😄